include/nfc/tnep/ch.h

```diff
@@ -0,0 +1,203 @@
+
+/**@brief Connection Handover Record structure. */
+struct nfc_tnep_ch_record {
+	/** Pointer to the first Alternative Carrier Record. */
+	const struct nfc_ndef_ch_ac_rec *ac;
+
+	/** Pointer to the first Carrier Record. */
+	const struct nfc_ndef_record_desc **carrier;
+
+	/** Count of the Alternative Carriers Records and Carrier Records. */
+	size_t count;
+
+	/** Connection Handover major version. */
+	uint8_t major_ver;
+
+	/** Connection Handover minor version. */
+	uint8_t minor_ver;
+};
+
+/**@brief Connection Handover Request Record strucure. */
+struct nfc_tnep_ch_request {
+	/** Pointer to the Collision Resolution Record data. */
+	const struct nfc_ndef_ch_cr_rec *cr;
+
+	/** Connection Handover Record data. */
+	struct nfc_tnep_ch_record ch_record;
+};
+
+/**@brief Connection Handover service callback structure. */
+struct nfc_tnep_ch_cb {
+	/**@brief The Connection Handover Request Message prepare callback.
+	 *
+	 * This callback is called every time when the NFC Forum Device
+	 * takes the Connection Handover Requester role and the
+	 * Connection Handover Message must be sent to the other
+	 * NFC Forum Device.
+	 *
+	 * @retval 0 If the operation was successful.
+	 *         Otherwise, a (negative) error code is returned.
+	 */
+	int (*request_msg_prepare)(void);
+
+	/**@brief The Connection Handover Request Message received callback.
+	 *
+	 * This callback is called always when NFC Forum Device which
+	 * supports TNEP received the Connection Handover Request Message.
+	 * After receiving this message, application shall respond with
+	 * the Handover Select Message by calling @ref nfc_tnep_ch_carrier_set.
+	 *
+	 * @param[in] ch_req Pointer to the Connection Handover Request
+	 *                   structure which contains the parsed Connection
+	 *                   Handover Request Message.
+	 *
+	 * @retval 0 If the operation was successful.
+	 *         Otherwise, a (negative) error code is returned.
+	 */
+	int (*request_msg_recv)(const struct nfc_tnep_ch_request *ch_req);
+
+	/**@brief The Connection Handover Select Message received callback.
+	 *
+	 * This callback is called always when the NFC Device receive
+	 * the Connection Select Message in response to the Connection
+	 * Handover Request message.
+	 *
+	 * @param[in] ch_select Pointer to Connection Handover structure
+	 *                      which contains the parsed Connection Handover
+	 *                      Select Message.
+	 * @param[in] inactive Indicates that all Alternative Carriers are
+	 *                     inactive. The Handover Request Message should
+	 *                     be sent again with only one Alternative Carrier
+	 *                     using @ref nfc_tnep_ch_carrier_set.
+	 *
+	 * @retval 0 If the operation was successful.
+	 *         Otherwise, a (negative) error code is returned.
+	 */
+	int (*select_msg_recv)(const struct nfc_tnep_ch_record *ch_select,
+			       bool inactive);
+
+#if defined(CONFIG_NFC_TNEP_TAG)
```

Comment:
I think this needs to be defined somewhere in the documentation build, otherwise it won't show in the published documentation.

@b-gent Is this correct, and do you know where this would have to be added?

Reply:
@KAGA164 I found it. It's in doc/nrf/nrf.doxyfile.in

Add CONFIG_NFC_TNEP_TAG to the list of PREDEFINED symbols.

Reply:
Thanks. Done